### ISSUE TITLE
Cannot find "name" file in device folder

### ISSUE THREAD

**Issue by schulmar** [matteocorti/check_lm_sensors#3]
When running `/usr/lib/monitoring-plugins/check_lm_sensors --list` I get the result:

```
LM_SENSORS UNKNOWN - Error reading /sys/class/hwmon/hwmon0/device/name: No such file or directory
```

Looking in the respective directories I see:

```
$ ls -l /sys/class/hwmon/hwmon0/device/
total 0
lrwxrwxrwx 1 root root    0 Nov  2 14:20 driver -> ../../../bus/platform/drivers/coretemp
-rw-r--r-- 1 root root 4096 Nov 19 13:16 driver_override
drwxr-xr-x 3 root root    0 Nov 19 13:20 hwmon
-r--r--r-- 1 root root 4096 Nov 19 13:16 modalias
drwxr-xr-x 2 root root    0 Nov 19 13:20 power
lrwxrwxrwx 1 root root    0 Nov  2 14:20 subsystem -> ../../../bus/platform
-rw-r--r-- 1 root root 4096 Nov 19 13:16 uevent
```

```
$ ls -l /sys/class/hwmon/hwmon0/
total 0
lrwxrwxrwx 1 root root    0 Nov  2 14:20 device -> ../../../coretemp.0
-r--r--r-- 1 root root 4096 Nov 19 13:16 name
drwxr-xr-x 2 root root    0 Nov 19 13:17 power
lrwxrwxrwx 1 root root    0 Nov 19 13:17 subsystem -> ../../../../../class/hwmon
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp2_crit
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp2_crit_alarm
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp2_input
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp2_label
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp2_max
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp3_crit
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp3_crit_alarm
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp3_input
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp3_label
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp3_max
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp4_crit
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp4_crit_alarm
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp4_input
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp4_label
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp4_max
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp5_crit
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp5_crit_alarm
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp5_input
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp5_label
-r--r--r-- 1 root root 4096 Nov 19 13:16 temp5_max
-rw-r--r-- 1 root root 4096 Nov 19 13:16 uevent
```

So it seems the `name` file has been moved up from the device subfolder.
Obviously the old location might still be in use somewhere else so both should be checked.

**Comment by matteocorti** [comment]
Dear Martin

as I do not have any single non virtualised Linux machine at disposal, it will be difficult to analyse the problem. If you or someone else are willing to patch the plugin I will be glad to integrate the correction.

Which kernel version are you using?

Matteo

**Comment by schulmar** [comment]
I am using Kernel 4.2.5-1-ARCH.

I got it to work on my machine by removing the device subfolder in the [code](https://github.com/matteocorti/check_lm_sensors/blob/master/check_lm_sensors#L393).
However, I am no Perl programmer and am not sure whether I could create a patch that would make it work on my machine while maintaining "backwards compatibility" to the previous versions.

**Comment by schulmar** [comment]
As you can see, I issued a pull request.
It works for my machine. 
Tests for other configurations would have to be done by different people.

**Comment by matteocorti** [comment]
Dear Martin,

many thanks!

Maybe I am doing something wrong but I see no pull requests.

Matteo

**Comment by schulmar** [comment]
Ahhh, I made the changes in the browser and github offered me to do the pull request right away.
I did not notice that it is against my fork of your repo.

I reissued a pull request against your master branch. I am sorry for the confusion.

**Comment by matteocorti** [comment]
I merged your request and published version 4.1.0. Many thanks!

**Comment by matteocorti** [comment]
Forgot to close the issue
